service-base/src/main/java/org/eclipse/hono/service/amqp/RequestResponseEndpoint.java

```diff
@@ -496,4 +527,32 @@ protected boolean isValidReplyToAddress(final ResourceIdentifier replyToAddress)
             return replyToAddress.getResourcePath().length >= 3;
         }
     }
+
+    /**
+     * Creates {@code DeliveryOptions} that contain the given {@code SpanContext}.
+     * <p>
+     * To be used when sending a message on the vert.x event bus.
+     *  
+     * @param spanContext The {@code SpanContext} (may be {@code null}).
+     * @return The {@code DeliveryOptions}.
+     */
+    protected final DeliveryOptions createEventBusMessageDeliveryOptions(final SpanContext spanContext) {
+        final DeliveryOptions deliveryOptions = new DeliveryOptions();
+        if (spanContext != null && !(spanContext instanceof NoopSpanContext)) {
+            final MultiMap multiMap = new CaseInsensitiveHeaders();
+            tracer.inject(spanContext, Format.Builtin.TEXT_MAP, new MultiMapInjectAdapter(multiMap));
+            deliveryOptions.setHeaders(multiMap);
+        }
+        return deliveryOptions;
+    }
+
+    /**
+     * Extracts a {@code SpanContext} out of the delivery annotations of the given {@code Message}.
+     * 
+     * @param msg The AMQP message.
+     * @return The extracted {@code SpanContext} (may be {@code null}).
```

Comment:
should document that msg must not be `null`

Reply:
I've adapted the javadoc accordingly.